src/api/InterestAPI.js

```diff
@@ -32,7 +32,14 @@ async function post(url, query, toJson = true) {
     }
 
     if (response.status !== 200) {
-        throw new APIError(response.statusText, response.status);
+        let errorMsg;
+        try {
+            const errorJson = response.json();
+            errorMsg = errorJson.error_code || response.statusText;
```

Comment:
Hvis jeg leser dette riktig så forventer den beskjeden i `"error_code"` eller `"statusText"` er det riktig? Kommer til å bruke `"error_code"` uansett men er bare nysgjerrig.

Reply:
`statusText` er fallback hvis det ikke er noe json der, og blir i dette tilfellet Bad Request.

Responsen vil alltid inneholde status=400 og statusText=Bad Request. I tillegg kan den jo også inneholde json i response body, og da forsøker vi å parse den for å se om det er en mer spesifikk feilmelding der

Reply:
`errorMsg = errorJson.error_code || response.statusText;`

er det samme som :
```
if (errorJson.error_code !== undefined)
   errorMsg = errorJson.error_code;
} else {
  errorMsg = response.statusText;
}
```

Reply:
Ja, det er slik du sier. Om `errorJson.error_code` ikke finnes, bruker man `statusText` i stedet